### PR TITLE
Include method definitions in FunctionData message

### PR DESCRIPTION
**Pull request by mwaskom**
Fixes CLI-275.

I'm not really sure how to write a unit test for this -- open to suggestions. See the ticket for a simple repro.

<details> <summary>Backward/forward compatibility checks</summary>

---

Check these boxes or delete any item (or this section) if not relevant for this PR.

- [x] Client+Server: this change is compatible with old servers
- [x] Client forward compatibility: this change ensures client can accept data intended for later versions of itself

Note on protobuf: protobuf message changes in one place may have impact to
multiple entities (client, server, worker, database). See points above.

---

</details>

## Changelog

- Fixed a bug introduced in v0.67.0 where it was impossible to call `modal.Cls` methods when passing a list of requested GPUs.

### CODE REVIEW

File: modal/functions.py [hunk 1]
```diff
@@ -886,6 +886,8 @@ async def _load(self: _Function, resolver: Resolver, existing_object_id: Optiona
                         is_method=function_definition.is_method,
                         use_function_id=function_definition.use_function_id,
                         use_method_name=function_definition.use_method_name,
+                        method_definitions=method_definitions,
```
Comment (irfansharif) [inline modal/functions.py:889]: [nit] Can we make this look like `method_definitions=function_definition.method_definitions`, ditto for the other field. If it helps, after the FunctionData migration, we'll be able to have just one form this in the client - i.e. filling out the FunctionData proto directly. 

In the interm - we have this code in the server. We could pull it into the client and use that instead:

https://github.com/modal-labs/modal/blob/159c468655850cd8f53d6506bc3138b9e29c4e58/server/modal_server/function_definition.py#L89.

**Review by irfansharif — APPROVED** [review]
🤦🏽

File: modal/functions.py (hunk 1 above)
Comment (mwaskom) [reply modal/functions.py]: Dumb on my part for not noticing this.

**Comment by mwaskom** [comment]
@prbot automerge

**Pull request merged**